ServiceWebsite/ServiceWebsite.UnitTests/Controllers/ParticipantControllerTests.cs

```diff
@@ -17,20 +18,73 @@ public class ParticipantControllerTests
             _controller.MockUserIdentity(Username);
         }
 
+        [Test]
+        public async Task should_return_badrequest_if_hearingId_is_empty()
+        {
+            var result = await _controller.UpdateSuitabilityAnswers(Guid.Empty, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
+            var badRequestresult = (BadRequestObjectResult)result;
+            Assert.AreEqual(400, badRequestresult.StatusCode);
+        }
+
         [Test]
         public async Task should_return_not_found_if_no_hearing_for_user_is_found()
         {
-            var result = await _controller.UpdateSuitabilityAnswers(_hearingId, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
-            var noContentResult = (NoContentResult)result;
-            Assert.AreEqual(204, noContentResult.StatusCode);
+            // given service returns
+            _hearingService.Setup(x => x.GetParticipantIdAsync(Username, _hearingId))
+                .ThrowsAsync(new NotFoundException("message"));
+            var result = (NotFoundObjectResult)await _controller.UpdateSuitabilityAnswers(_hearingId, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
+            Assert.AreEqual(404, result.StatusCode);
+        }
+
+        [Test]
+        public async Task should_return_unauthorized_for_hearing_the_user_is_not_participant_in()
+        {
+            // given service throws
+            _hearingService.Setup(x => x.GetParticipantIdAsync(Username, _hearingId)).Returns(Task.FromResult((Guid?)Guid.Empty));
+            var result = (UnauthorizedObjectResult)await _controller.UpdateSuitabilityAnswers(_hearingId, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
+            Assert.AreEqual(result.Value, $"User is not a participant of hearing with id '{_hearingId}'");
+        }
+
+        [Test]
+        public async Task should_return_badrequest_if_question_key_format_is_incorrect()
+        {
+            var answers = new List<HearingSuitabilityAnswer> { new HearingSuitabilityAnswer() { QuestionKey = "incorrect_key", Answer = "Test Answer", ExtendedAnswer = "Test Extended Answer" } };
+            _hearingService.Setup(x => x.GetParticipantIdAsync(Username, _hearingId)).Returns(Task.FromResult((Guid?)Guid.NewGuid()));
+
+            var result = (BadRequestObjectResult)await _controller.UpdateSuitabilityAnswers(_hearingId, answers);
+            Assert.AreEqual(400, result.StatusCode);
+        }
+
+        [Test]
+        public async Task should_return_badrequest_if_answers_length_is_zero()
+        {
+           _hearingService.Setup(x => x.GetParticipantIdAsync(Username, _hearingId)).Returns(Task.FromResult((Guid?)Guid.NewGuid()));
+
+            var result = (BadRequestObjectResult)await _controller.UpdateSuitabilityAnswers(_hearingId, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
+            Assert.AreEqual(400, result.StatusCode);
```

Comment:
Check the response message here as well, otherwise, we might accidentally be testing another case where some other input fails with a 400 response.

Reply:
Resolved